flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -166,21 +166,40 @@ struct CharacterOpsBuilderImpl {
       return fir::ReferenceType::get(getCharacterType());
     }
 
-    bool isConstant() const { return data.getType().isa<fir::SequenceType>(); }
+    bool needToMaterialize() const {
+      return data.getType().isa<fir::SequenceType>() ||
+             data.getType().isa<fir::CharacterType>();
+    }
+
+    std::optional<fir::SequenceType::Extent> getCompileTimeLength() const {
+      auto type = data.getType();
+      if (type.isa<fir::CharacterType>())
+        return 1;
+      if (auto refType = type.dyn_cast<fir::ReferenceType>())
+        type = refType.getEleTy();
+      if (auto seqType = type.dyn_cast<fir::SequenceType>()) {
+        auto shape = seqType.getShape();
+        assert(shape.size() == 1 && "only scalar character supported");
+        return shape[0];
```

Comment:
In general, this could contain the "unknown extent" value, which is negative.

Reply:
Thanks !

Reply:
If OK to merge without the fix ? it is in my len_trim branch but I do not want to push all these changes yet.

Reply:
Yes